arches/app/views/api/controlled_lists.py

```diff
@@ -0,0 +1,424 @@
+                [
+                    f"controlled_list_items{'__children' * i}",
+                    f"controlled_list_items{'__children' * i}__controlled_list_item_values",
+                    f"controlled_list_items{'__children' * i}__controlled_list_item_images",
+                    f"controlled_list_items{'__children' * i}__controlled_list_item_images__controlled_list_item_image_metadata",
+                ]
+            )
+    return terms
+
+
+@method_decorator(
+    group_required("RDM Administrator", raise_exception=True), name="dispatch"
+)
+class ControlledListsView(View):
+    def get(self, request):
+        """Returns either a flat representation (?flat=true) or a tree (default)."""
+        lists = (
+            ControlledList.objects.annotate_node_fields(
+                node_ids="pk",
+                node_names="name",
+                nodegroup_ids="nodegroup_id",
+                graph_ids="graph_id",
+                graph_names="graph__name",
+            )
+            .order_by("name")
+            .prefetch_related(*prefetch_terms(request))
+        )
+
+        flat = str_to_bool(request.GET.get("flat", "false"))
+        permitted = [
+            ng.pk for ng in get_nodegroups_by_perm(request.user, "read_nodegroup")
+        ]
+        serialized = [
+            obj.serialize(flat=flat, permitted_nodegroups=permitted) for obj in lists
+        ]
+
+        return JSONResponse({"controlled_lists": serialized})
+
+
+@method_decorator(
+    group_required("RDM Administrator", raise_exception=True), name="dispatch"
+)
+class ControlledListView(View):
+    def get(self, request, **kwargs):
+        """Returns either a flat representation (?flat=true) or a tree (default)."""
+        list_id = kwargs.get("id")
+        try:
+            lst = ControlledList.objects.prefetch_related(*prefetch_terms(request)).get(
+                pk=list_id
+            )
+        except ControlledList.DoesNotExist:
+            return JSONErrorResponse(status=HTTPStatus.NOT_FOUND)
+
+        flat = str_to_bool(request.GET.get("flat", "false"))
+        permitted = [
+            ng.pk for ng in get_nodegroups_by_perm(request.user, "read_nodegroup")
+        ]
+        serialized = lst.serialize(flat=flat, permitted_nodegroups=permitted)
+
+        return JSONResponse(serialized)
+
+    def add_new_list(self, name):
+        lst = ControlledList(name=name)
+        try:
+            lst.full_clean()  # applies default name
+        except ValidationError as ve:
+            return JSONErrorResponse(
+                message="\n".join(ve.messages), status=HTTPStatus.BAD_REQUEST
+            )
+        lst.save()
+        return JSONResponse(lst.serialize(), status=HTTPStatus.CREATED)
+
+    def post(self, request, **kwargs):
+        data = JSONDeserializer().deserialize(request.body)
+        name = data.get("name", None)
+        if "id" not in kwargs:
+            return self.add_new_list(name)
+
+        raise NotImplementedError
```

Comment:
I removed the prior functionality that let you post a huge payload of absolutely everything pertaining to a list (items, values, notes, images), given that I'd already moved away from using it in the frontend to improve performance. That meant I could say goodbye to the `handle_items` helper we discussed before. We can bring it back if we have a use case for it.